### PR TITLE
[DOCS]: consolidate doc site content simplify navbar

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
Closes https://github.com/apache/arrow-datafusion/issues/5935
Related to https://github.com/apache/arrow-datafusion/issues/1814 and https://github.com/apache/arrow-datafusion/issues/5501

# Rationale for this change
The main page / index of the https://arrow.apache.org/datafusion/ site is somewhat disorganized, redundant and has so many entries it is causing issues such as https://github.com/apache/arrow-datafusion/issues/5935

Here is a screenshot of the current site:
Screenshot 2023-04-11 at 7 20 00 AM

# What changes are included in this PR?
1. Consolidate some top level pages into lower level pages ("improve the site navigation") to reduce the size of the initial index and organize the content better. 
2. Various small improvements

# Are these changes tested?

I rendered the site locally and it looks better to me:

Screenshot 2023-04-11 at 7 22 45 AM

# Are there any user-facing changes?

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -50,22 +49,17 @@ community.
 
    user-guide/introduction
    user-guide/example-usage
-   user-guide/users
```
Comment (alamb) [inline docs/source/index.rst:53]: I consolidated the content of these pages into other pages

File: docs/source/index.rst [hunk 2]
```diff
@@ -50,22 +49,17 @@ community.
...
-   user-guide/comparison
-   user-guide/integration
-   user-guide/library
    user-guide/cli
    user-guide/dataframe
    user-guide/expressions
    user-guide/sql/index
    user-guide/configs
    user-guide/faq
-   Rust Crate Documentation <https://docs.rs/crate/datafusion/>
```
Comment (alamb) [inline docs/source/index.rst:63]: This was redundant with the crates.io link above

File: docs/source/index.rst [hunk 3]
```diff
@@ -50,22 +49,17 @@ community.
...
 
 .. _toc.contributor-guide:
 
 .. toctree::
-   :maxdepth: 2
```
Comment (alamb) [inline docs/source/index.rst:68]: This stops listing H2 headings ( `##`) on the main table of contents

File: docs/source/user-guide/example-usage.md [hunk 4]
```diff
@@ -141,3 +141,112 @@ async fn main() -> datafusion::error::Result<()> {
...
+  ctx.register_csv("test", "<PATH_TO_YOUR_CSV_FILE>", CsvReadOptions::new()).await?;
+
+  // create a plan to run a SQL query
+  let df = ctx.sql("SELECT * FROM test").await?;
+
+  // execute and print results
+  df.show().await?;
+  Ok(())
+}
+```
```
Comment (Jefffrey) [inline docs/source/user-guide/example-usage.md:195]: example feels kinda redundant compared with example code in above sections

File: docs/source/user-guide/faq.md [hunk 5]
```diff
@@ -29,3 +29,37 @@ model and computational kernels. It is designed to run within a single process,
...
+- [DuckDB](http://www.duckdb.org) is an open source, in process analytic database.
+  Like DataFusion, it supports very fast execution, both from its custom file format
+  and directly from parquet files. Unlike DataFusion, it is written in C/C++ and it
+  is primarily used directly by users as a serverless database and query system rather
+  than as a library for building such database systems.
+
+- [Polars](http://pola.rs): Polars is one of the fastest DataFrame
+  libraries at the time of writing. Like DataFusion, it is also
+  written in Rust and uses the Apache Arrow memory model, but unlike
+  DataFusion it does not provide SQL nor as many extension points.
```
Comment (Jefffrey) [inline docs/source/user-guide/faq.md:54]: change to https url

also i think polars might support sql now, according to their doc: https://pola-rs.github.io/polars-book/user-guide/sql.html

File: docs/source/user-guide/faq.md [hunk 6]
```diff
@@ -29,3 +29,37 @@ model and computational kernels. It is designed to run within a single process,
...
+
+- [Facebook Velox](https://engineering.fb.com/2022/08/31/open-source/velox/)
```
Comment (Jefffrey) [inline docs/source/user-guide/faq.md:56]: could switch to github link: https://github.com/facebookincubator/velox since this link seems dead

**Review by Jefffrey — COMMENTED** [review]
Hope you don't mind but took the opportunity to review the docs and point out a few parts that are outdated/could be improved

File: docs/source/contributor-guide/architecture.md [hunk 7]
```diff
@@ -20,7 +20,8 @@
 # Architecture
 
 DataFusion's code structure and organization is described in the
-[Crate Documentation], to keep it as close to the source as
-possible.
+[crates.io documentation], to keep it as close to the source as
+possible. You can find the most up to date version in the [source code].
```
Comment (waynexia) [inline docs/source/contributor-guide/architecture.md:24]: What do you think about hosting the latest document generated from the source code on github pages (or other static page hoster)? Like [greptimedb.rs](https://greptimedb.rs) which is generated from https://github.com/GreptimeTeam/greptimedb/deployments/activity_log?environment=github-pages

**Review by waynexia — COMMENTED** [review]
Thanks for bringing up this topic. I also left some unrelated comment 🙇‍♂️

File: docs/source/user-guide/faq.md (hunk 5 above)
Comment (alamb) [reply docs/source/user-guide/faq.md:54]: Yes I think you are correct.

**Comment by alamb** [comment]
> Hope you don't mind but took the opportunity to review the docs and point out a few parts that are outdated/could be improved

Not at all - thank you @Jefffrey  and thank you @waynexia

File: docs/source/user-guide/example-usage.md [hunk 8]
```diff
@@ -141,3 +141,112 @@ async fn main() -> datafusion::error::Result<()> {
...
+To get started, add the following to your `Cargo.toml` file:
+
+```toml
+[dependencies]
+datafusion = "11.0"
+```
+
+## Create a main function
+
+Update the main.rs file with your first datafusion application based on [Example usage](https://arrow.apache.org/datafusion/user-guide/example-usage.html)
```
[the remark this answers is not in the document]
Comment (alamb) [reply docs/source/user-guide/example-usage.md]: Yes, that is a good a catch. This whole page has some non trivial redundancy. I will try and fix it up

File: docs/source/user-guide/example-usage.md (hunk 4 above)
Comment (alamb) [reply docs/source/user-guide/example-usage.md]: I agree -- removed

File: docs/source/user-guide/example-usage.md [hunk 9]
```diff
@@ -141,3 +141,112 @@ async fn main() -> datafusion::error::Result<()> {
...
+This crate is tested with the latest stable version of Rust. We do not currently test against other, older versions of the Rust compiler.
+
+## Optimized Configuration
+
+For an optimized build several steps are required. First, use the below in your `Cargo.toml`. It is
+worth noting that using the settings in the `[profile.release]` section will significantly increase the build time.
+
+```toml
+[dependencies]
+datafusion = { version = "11.0" , features = ["simd"]}
```
[the remark this answers is not in the document]
Comment (alamb) [reply docs/source/user-guide/example-usage.md]: Maybe we could update the script here to automatically clean it up: https://github.com/apache/arrow-datafusion/blob/main/dev/update_datafusion_versions.py

File: docs/source/user-guide/faq.md [hunk 10]
```diff
@@ -29,3 +29,37 @@ model and computational kernels. It is designed to run within a single process,
...
+
+1. Targeted at developers, rather than end users / data scientists.
+2. Designed to be embedded, rather than a complete file based SQL system.
+3. Governed by the [Apache Software Foundation](https://www.apache.org/) process, rather than a single company or individual.
+4. Implemented in `Rust`, rather than `C/C++`
+
+Here is a comparison with similar projects that may help understand
+when DataFusion might be be suitable and unsuitable for your needs:
+
+- [DuckDB](http://www.duckdb.org) is an open source, in process analytic database.
```
[the remark this answers is not in the document]
Comment (alamb) [reply docs/source/user-guide/faq.md]: in c18786332

**Comment by alamb** [comment]
Ok, I think this is better than what is on main so I will merge it in. We clearly have a ways to go to have wonderful docs

File: docs/source/contributor-guide/architecture.md (hunk 7 above)
Comment (alamb) [reply docs/source/contributor-guide/architecture.md:24]: Filed https://github.com/apache/arrow-datafusion/issues/5981

File: docs/source/user-guide/example-usage.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (waynexia) [reply docs/source/user-guide/example-usage.md]: Sounds good, I filed #5983

**Pull request merged**